belief_manager.py

```diff
@@ -0,0 +1,578 @@
+    past_lob = past_data[0]
+    past_x = past_data[2]
+    past_y = past_data[3]
+    past_slope = np.tan(past_lob)
+    past_intercept = past_y - past_slope * past_x
+
+    # Calculate where they intersect.
+    intersect_x = (current_intercept - past_intercept) / \
+                  (past_slope - current_slope)
+    intersect_y = past_slope * intersect_x + past_intercept
+
+    return (intersect_x, intersect_y)
+
+  def _single_point_strength_distance(self, position, lob, strength):
+    """ Does the distance-from-strength calculation for a single point.
+    Args:
+      position: The drone position to use in the calculation. (X, Y)
+      lob: The LOB to the transmitter that we are using.
+      strength: The strength associated with the LOB reading.
+    Returns:
+      A tuple containing the X and Y coordinates of the calculated position. """
+    # TODO(danielp): Implement this function for real.
+    logger.critical("_distance_from_strength not implemented!")
+
+    # For now, strengths have a 1-to-1 correlation with distance.
+    position_x = position[self._X] + strength * np.cos(lob)
+    position_y = position[self._Y] + strength * np.sin(lob)
+
+    return (position_x, position_y)
+
+  def _calculate_distance(self, readings):
+    """ Use the best method to calculate distance, given the circumstances.
+    readings: Associated LOB readings.
+    Returns:
+      A dict of tuples, with each tuple constaining the estimated x and y
+      position of the transmitter. The keys are the indices of the corresponding
+      transmitters in the state. """
+    strength_readings = {}
+    past_measurement_transmitters = []
+    for transmitter, reading in readings.iteritems():
+      current_position = self._filter.position()
+      if transmitter - 4 >= len(self._past_lobs):
+        # Use strength, this transmitter is too new.
+        strength_readings[transmitter] = reading
+        continue
+
+      past_x = self._past_lobs[transmitter - 4][2]
+      past_y = self._past_lobs[transmitter - 4][3]
+
+      past_measurement_transmitters.append(transmitter)
+
+    distances = self._distance_from_strength(strength_readings)
+    logger.debug("Distance: Using past data for transmitters: %s" % \
+                 (past_measurement_transmitters))
+    distances2 = self._distance_from_past_states(past_measurement_transmitters)
+    # Combine them.
+    for transmitter, position in distances2.iteritems():
+      distances[transmitter] = distances2[transmitter]
+
+    return distances
+
+  def _transmitter_error_regions(self, stddevs, existing, new):
+    """ Calculates error regions for the transmitters that we can currently see,
+    and saves them.
+    Args:
+      stddevs: The z-score to use when calculating the error regions.
+      existing: Associated readings for existing transmitters.
+      new: Readings for new transmitters. """
+    num_points = 40 + 10 * self._filter.number_of_transmitters()
```

Comment:
Yeah, can't argue with that... I think it's mainly lacking a comment, so I've added one.